### PR TITLE
[FLINK-30100][checkpointing] Remove the unused CheckpointFailureReason

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What is the purpose of the change

Some CheckpointFailureReasons have not been used for a long time and should be removed. E.g: 

- TOO_MANY_CONCURRENT_CHECKPOINTS
- CHECKPOINT_DECLINED_TASK_NOT_CHECKPOINTING
- CHECKPOINT_DECLINED_ALIGNMENT_LIMIT_EXCEEDED
- JOB_FAILURE

## Brief change log

- The first commit: Remove some unused exceptions and refactor some code of CheckpointCoordinatorTest
- The second commit: Remove deprecated getCheckpointId of PendingCheckpoint
- The third commit: Migrate checkpoint coordinator related tests to junit5
- The fourth commit: Remove the unused CheckpointFailureReason

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by 1996fanrui** [comment]
Hi @pnowojski please help take a look, thanks. It's also a cleanup PR.

**Review by pnowojski — CHANGES_REQUESTED** [review]
LGTM % one comment. Thanks for the clean up

File: flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java [hunk 1]
```diff
@@ -3837,14 +3829,6 @@ public void testBaseLocationsNotInitialized() throws Exception {
...
                         .setTransitToRunning(false)
                         .build(EXECUTOR_RESOURCE.getExecutor());
-        CheckpointCoordinator checkpointCoordinator =
-                new CheckpointCoordinatorBuilder()
-                        .setCheckpointCoordinatorConfiguration(
-                                CheckpointCoordinatorConfiguration.builder()
-                                        .setCheckpointInterval(Long.MAX_VALUE)
-                                        .build())
-                        .setCheckpointStorage(new FsStateBackend(checkpointDir.toURI()))
-                        .build(graph);
```
[the remark this answers is not in the document]
Comment (1996fanrui) [reply flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java:3847]: You are right, It's my mistake.  I see this checkpointCoordinator isn't used, so removed it. 

It's covered, and I removed the `CheckpointCoordinator checkpointCoordinator =` due to it's not necessary.

BTW, I did some extra work in this PR during fix this comment.

I migrate checkpoint coordinator related tests to junit5 in the 3rd commit.

**Review by pnowojski — APPROVED** [review]
Thanks for the update, LGTM

**Comment by 1996fanrui** [comment]
@pnowojski Thanks for your review!

**Pull request merged**